### PR TITLE
Url converters tests

### PR DESCRIPTION
**Pull request by adrianmroz**

### CODE REVIEW

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+      const expected = essence();
+      assertEqlEssence(result, expected);
+    });
+  });
+
+  describe("Splits", () => {
+    it("reads string split", () => {
+      const result = toEssence(viewDefinition({
+        splits: [SplitDefinitionFixtures.stringSplitDefinition("string_a", "string_a")],
+        visualization: TABLE_MANIFEST.name
```
Comment (erykpiast) [inline src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: I'd descript the relation between `TABLE_MANIFEST.name` and splits in input to `TABLE_MANIFEST` in the output.

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+      const result = toEssence(viewDefinition());
+      const expected = essence();
+      assertEqlEssence(result, expected);
+    });
+  });
+
+  describe("Splits", () => {
+    it("reads string split", () => {
+      const result = toEssence(viewDefinition({
+        splits: [SplitDefinitionFixtures.stringSplitDefinition("string_a", "string_a")],
```
Comment (erykpiast) [inline src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: I'd hide the boilerplate like `SplitDefinitionFixtures.stringSplitDefinition` and `Splits.fromSplit` in some local helper. There is a lot of noise in these tests.

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts [hunk 3]
```diff
@@ -0,0 +1,261 @@
...
+        visualization: TABLE_MANIFEST.name
+      }));
+      const expected = essence({
+        splits: Splits.fromSplit(SplitFixtures.stringSplitCombine("string_a", "string_a", SortDirection.ascending, 10)),
+        visualization: TABLE_MANIFEST
+      });
+      assertEqlEssence(result, expected);
+    });
+
+    it("reads time split", () => {
```
Comment (erykpiast) [inline src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: move test for each kind of split to separate `describe` blocks

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+      const expected = essence({
+        splits: Splits.fromSplit(SplitFixtures.timeSplitCombine("time", "PT2M")),
+        visualization: TABLE_MANIFEST
+      });
+      assertEqlEssence(result, expected);
+    });
+
+    it("reads number split", () => {
+      const result = toEssence(viewDefinition({
+        splits: [SplitDefinitionFixtures.numberSplitDefinition("numeric", 100, "numeric")],
```
Comment (erykpiast) [inline src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: specify sorting here explicitly or unify defaults

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts [hunk 5]
```diff
@@ -0,0 +1,261 @@
...
+      const result = toEssence(viewDefinition({
+        series: [SeriesDefinitionFixtures.quantileSeriesDefinition("quantile", 90)]
+      }));
+      const expected = essence({
+        series: SeriesList.fromSeries([SeriesFixtures.quantileSeries("quantile", 90)])
+      });
+      assertEqlEssence(result, expected);
+    });
+
+    it("infers quantile from simple definition for measure with quantile expression", () => {
```
Comment (erykpiast) [inline src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: `... for backward compatibility`

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts [hunk 6]
```diff
@@ -0,0 +1,261 @@
...
+      const expected = essence({
+        splits: Splits.fromSplit(SplitFixtures.stringSplitCombine("string_a", "string_a")),
+        visualization: TABLE_MANIFEST
+      });
+      assertEqlEssence(result, expected);
+    });
+  });
+
+  describe("Series", () => {
+    it("reads simple series", () => {
```
Comment (erykpiast) [inline src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: separate `describe` blocks for different series kinds

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (adrianmroz) [reply src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: should unify default values for fixtures

File: src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts (hunk 1 above)
Comment (adrianmroz) [reply src/common/view-definitions/version-4/view-definition-converter-4.mocha.ts]: And add few cases for `visResolve` edge cases when requested visualisation isn't fitted to the rest of parameters.

**Pull request merged**